File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPSwimming.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPSwimming implements PlayerStateProperty<Boolean> {
    @Override
    public String getConfigName() {
        return "swimming";
    }

    @Override
    public Boolean getDefaultSerializedProperty() {
        return false;
    }

    @Override
    public Boolean getSerializedPropertyFromPlayer(Player player) {
        return player.isSwimming();
    }

    @Override
    public void restoreProperty(Player player, Boolean swimming) {
        player.setSwimming(swimming);
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPSwimming.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPSwimming extends PlayerStateProperty[Boolean]:
  override def getConfigName: String = "swimming"

  override def getDefaultSerializedProperty: Boolean = false

  override def getSerializedPropertyFromPlayer(player: Player): Boolean =
    player.isSwimming

  override def restoreProperty(player: Player, swimming: Boolean): Unit =
    player.setSwimming(swimming)